hmda/src/main/scala/hmda/validation/rules/lar/quality/_2025/Q616_3.scala

```diff
@@ -0,0 +1,24 @@
+package hmda.validation.rules.lar.quality._2025
+
+import hmda.model.filing.lar.LoanApplicationRegister
+import hmda.validation.dsl.ValidationResult
+import hmda.validation.rules.EditCheck
+import hmda.validation.dsl.PredicateHmda._
+import hmda.validation.dsl.PredicateCommon._
+import hmda.validation.dsl.PredicateSyntax._
+
+import scala.util.Try
+
+
+object Q616_3 extends EditCheck[LoanApplicationRegister] {
+  override def name: String = "Q616-3"
+
+  override def parent: String = "Q616"
+
+  override def apply(lar: LoanApplicationRegister): ValidationResult = {
+    val discountPoints = BigDecimal(Try(lar.loanDisclosure.discountPoints.toDouble).getOrElse(0.0))
```

Comment:
we can also do `when(lar.loanDisclosure.discountPoints is numeric)` rather than the .toDouble.getOrElse

Reply:
This isn't checking to see if its numeric but rather ensuring we're comparing BigDecimal to BigDecimal a few lines after this.